### PR TITLE
LOOP-1964: Watch: For pre-meal bolus, clear the pre-meal override while calculating recommendation

### PR DESCRIPTION
**Pull request by rickpasetto**
Not sure this is the best solution here, but it seems to be the least intrusive.  The downside here is that the pre-meal icon flashes while the app on the phone calculates the bolus recommendation for the watch.

### CODE REVIEW

File: Loop/Managers/WatchDataManager.swift [hunk 1]
```diff
@@ -253,13 +253,16 @@ final class WatchDataManager: NSObject {
                 context.glucoseTrendRawValue = trend.rawValue
             }
             
+            var savedPreMealOverride: TemporaryScheduleOverride?
             if let potentialCarbEntry = potentialCarbEntry {
+                savedPreMealOverride = loopManager.settings.preMealOverride
+                loopManager.settings.preMealOverride = nil
```
Comment (ps2) [inline Loop/Managers/WatchDataManager.swift:259]: Hmm, I thought your fix in your other PR was to make LoopDataManager not consider the pre-meal override when there was a potentialCarbEntry. Shouldn't that work here as well?

Comment (rickpasetto) [reply Loop/Managers/WatchDataManager.swift:259]: The other PR only does this via the `BolusEntryViewModel` (which effectively does this... set `loopManager.settings.preMealOverride` to nil), but the watch does not go through that code path.

Comment (ps2) [reply Loop/Managers/WatchDataManager.swift:259]: Ah, ok, I don't think I read it closely enough then. What I thought was happening was that you changed LDM to use the `potentialCarbEntry` as a flag to determine whether to use preMealOverride.  I do think that's a great idea. 

We already pass `potentialCarbEntry` to the top level `recommendBolus` method. It then calls `recommendBolusValidatingDataRecency`, which would need to change to take a flag about whether the recommendation should consider premeal.  That same flag could be passed to `recommendBolus(forPrediction: predictedGlucose)`, which is where the target range is looked up. That lookup could be modified to use something like `glucoseTargetRangeScheduleApplyingNonPreMealOverrideIfActive`, and then you'd get a recommendation without pre-meal affecting things. 

This approach would solve the issue for both watch and BolusEntryView without needing to do any temporary state changes.

Comment (rickpasetto) [reply Loop/Managers/WatchDataManager.swift:259]: > Ah, ok, I don't think I read it closely enough then. What I thought was happening was that you changed LDM to use the potentialCarbEntry as a flag to determine whether to use preMealOverride. I do think that's a great idea.

^ This turned out to be too difficult to achieve without massive refactoring.  We discussed this, remember? That's why I went the route I did with `BolusEntryViewModel`.

Comment (ps2) [reply Loop/Managers/WatchDataManager.swift:259]: What else would need to be refactored? I do remember discussing a path similar to this, but I think I was sidetracked by thinking that the stores' idea of overrides would need to be changed, but that's not the case. Looking at the code now, I believe it's just adding that flag to those two methods, and calling a simplified version of glucoseTargetRangeScheduleApplyingOverrideIfActive (that doesn't consider pre-meal) if the flag is true.

**Review by ps2 — COMMENTED** [review]
I see you requested another review, but I don't see the changes we discussed. Guessing the request was on accident; I'm just responding with a comment here to clear this out of my list. :). Please re-request when you'd like me to review.

**Comment by rickpasetto** [comment]
@ps2 Whoops, yeah that request-for-review was done before our discussion (sigh -- the "race conditions" inherent in async comms on multiple channels!)...  I will "close" this PR for now.  Thanks!

**Pull request closed without merging**